Compile with clang/llvm

- Yuni defines weird compile flags, we'll need to clean them up
- It is impossible to build the project in Mac OS, since it uses clang
- Compare build/execution time

To use clang:
export CC=/usr/bin/clang
export CXX=/usr/bin/clang++
Then build in a new folder

Yuni compilation errors:
src/ext/yuni/src/yuni/core/event/../../core/string/iterator.inc.hpp:192:27: error: use of overloaded operator '!=' is ambiguous (with operand types 'UTF8::Char' and 'const char')
src/ext/yuni/src/yuni/uuid/../core/system/stdint.h:141:2: error: expected ';' after struct} __float128

They're either probably flag errors, or real code problem. Clang is stricter then gcc for some type comparison
https://stackoverflow.com/questions/25745759/clang-vs-gcc-vs-msvc-template-conversion-operator-which-compiler-is-right/25758367#25758367
https://stackoverflow.com/questions/25110429/ambiguous-operator-overload-on-clang

--- a/src/ext/yuni/src/yuni/core/string/iterator.inc.hpp
+++ b/src/ext/yuni/src/yuni/core/string/iterator.inc.hpp
@@ -173,7 +173,7 @@ struct Model
         {
             do
             {
-                if (pChar == ' ' || pChar == '\t' || pChar == '\n' || pChar == '\r')
+                if (pChar == (uint)' ' || pChar == (uint)'\t' || pChar == (uint)'\n' || pChar == (uint)'\r')
                 {
                     forward();
                     if (pOffset >= end)
@@ -189,7 +189,7 @@ struct Model
         {
             do
             {
-                if (pChar != c)
+                if (pChar != (uint)c)
                 {
                     forward();
                     if (pOffset > endOffset)

--- a/src/ext/yuni/src/yuni/core/system/stdint.h
+++ b/src/ext/yuni/src/yuni/core/system/stdint.h
@@ -135,8 +135,8 @@ typedef unsigned int uint;
 #ifdef YUNI_OS_CLANG
 /* Try to fix issues with clang which does not yet support __float128, and produces
 ** compilation error in the STL */
-typedef struct
-{
-    double x, y;
-} __float128;
+/* typedef struct __float128 */
+/* { */
+/*     double x, y; */
+/* } __float128; */
 #endif

--- a/src/ext/yuni/src/yuni/core/system/stdint.h
+++ b/src/ext/yuni/src/yuni/core/system/stdint.h
@@ -131,12 +131,3 @@ typedef double yfloat64;
 /*! Convenient typedef around unsigned int */
 typedef unsigned int uint;
 #endif
-
-#ifdef YUNI_OS_CLANG
-/* Try to fix issues with clang which does not yet support __float128, and produces
-** compilation error in the STL */
-/* typedef struct __float128 */
-/* { */
-/*     double x, y; */
-/* } __float128; */
-#endif

--- a/src/ui/simulator/toolbox/create.h
+++ b/src/ui/simulator/toolbox/create.h
@@ -44,7 +44,7 @@ namespace Component
 ** \param object  A popinter to the object which will receive the onClick event
 ** \param method  The method to bind for the onClick event
 */
-template<class T, class StringT, class UserDataT = void*>
+template<class T, class StringT, class UserDataT>
 wxButton* CreateButton(wxWindow* parent,
                        const StringT& caption,
                        T* object = NULL,

--- a/src/ui/simulator/toolbox/create.hxx
+++ b/src/ui/simulator/toolbox/create.hxx
@@ -64,7 +64,7 @@ namespace Antares
 {
 namespace Component
 {
-template<class T, class StringT, class UserDataT = void*>
+template<class T, class StringT, class UserDataT>
 wxButton* CreateButton(wxWindow* parent,
                        const StringT& caption,
                        T* object,

